src/pyfaf/repos/dnf.py

```diff
@@ -87,34 +94,38 @@ def list_packages(self, architectures):
         release, arch, srpm_name, type, filename, url items.
         """
 
-        self.yum_base.arch.archlist = architectures
-
         result = []
         try:
-            packagelist = self.yum_base.doPackageLists('all', showdups=True)
-        except yum.Errors.RepoError as err:
+            packagelist = self.dnf_base.sack.query().filterm(arch=architectures)
+        except dnf.exceptions.RepoError as err:
             self.log_error("Repository listing failed: '{0}'".format(err))
             return result
 
-        pkgs = (packagelist.available +
-                packagelist.old_available +
-                packagelist.reinstall_available)
+# Get available packages
+###############################################################################
+# TODO: yum added also packagelist.old_available and
+# packagelist.reinstall_available which is missing in dnf
+###############################################################################
+        pkgs = packagelist.available()
 
         for package in pkgs:
             pkg = dict(name=package.name,
-                       base_package_name=package.base_package_name,
```

Comment:
keep the base_package_name, it is needed on other places in the code (see my other comments). Don't worry if it will be the same as `name`.